inst/scripts/2.4.0/make-hermann-spermatogenesis-data.Rmd

```diff
@@ -0,0 +1,205 @@
+---
+title: Processing the Hermann spermatogenesis dataset for RNA velocity analysis
+author: Charlotte Soneson
+date: "June 5, 2020"
+output:
+  BiocStyle::html_document:
+    titlecaps: false
+    toc_float: true
+bibliography: ../ref.bib
+---
+
+```{r style, echo=FALSE, results='hide', message=FALSE}
+library(BiocStyle)
+knitr::opts_chunk$set(error=FALSE, message=FALSE, warning=FALSE, eval=FALSE)
+```
+
+# Loading required R packages
+
+```{r}
+suppressPackageStartupMessages({
+    library(Biostrings)
+    library(BSgenome)
+    library(eisaR)
+    library(GenomicFeatures)
+    library(SummarizedExperiment)
+    library(tximeta)
+    library(rjson)
+    library(SingleCellExperiment)
+    library(dplyr)
+})
+```
+
+# Setting the working directory
+
+```{r}
+workdir <- tempdir()
+Sys.setenv(workdir = workdir)
+```
+
+# Downloading the reference sequences
+
+In order to perform the quantification of spliced and unspliced abundances with alevin [@srivastava2019alevin], we need to first prepare a suitable set of reference features (transcript isoform and intron sequences) to generate an index from. 
+We start by downloading the mouse reference genome and annotation gtf file from GENCODE.
+
+```{bash}
+wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M24/gencode.vM24.annotation.gtf.gz -P $workdir
+wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M24/GRCm38.primary_assembly.genome.fa.gz -P $workdir
+gunzip $workdir/GRCm38.primary_assembly.genome.fa.gz
+```
+
+# Extracting the transcript and intron sequences
+
+Next, we extract a GRanges object containing the genomic coordinates of each annotated transcript and intron. 
+
+```{r}
+gtf <- file.path(workdir, "gencode.vM24.annotation.gtf.gz")
```

Comment:
If you're going to read in the GTF file anyway, it may be simpler and also more consistent to convert an EnsDB object from **AnnotationHub** into a GTF file. This will allow the R code to entirely avoid the need for mappings.

Reply:
I'm not sure I see why this would be more consistent. The annotation files used here are all from the same Gencode release, so they are internally consistent. The corresponding EnsDb from AnnotationHub (release 99) does not contain exactly the same set of features as the Gencode GTF file. Also not exactly sure what you mean by mappings here, but there are lots of features (specifically, the expanded introns) that will be quantified but that are not contained in the EnsDb object.

Reply:
I was referring to consistency with the `ensembl=TRUE` mode that I have for a lot of the other functions. If it was already in Ensembl, the getter wouldn't need to have this option. Usually I don't have a choice because I have to work with whatever the authors emitted, but given that we're actually regenerating the counts here, we have an opportunity to avoid the mapping.

Unless there's a compelling reason to use GENCODE.

Reply:
Aha. Well, the IDs are already Ensembl IDs also in Gencode. I find Gencode more compelling since the provided transcriptome fasta and the GTF files are consistent (=contain the same set of transcripts) and thus typically cause less confusion.